flock_agent/daemon/daemon.py

```diff
@@ -357,3 +361,94 @@ def log(self, request, response, time):
         # Change permissions of unix socket
         os.chmod(self.unix_socket_path, 0o660)
         os.chown(self.unix_socket_path, 0, self.gid)
+
+    async def autoupdate_loop(self):
+        # Autoupdate is only available for macOS right now; Linux uses package managers
+        if Platform.current() != Platform.MACOS:
+            return
+
+        while True:
+            self.c.log("Daemon", "autoupdate_loop", "Checking for updates")
+
+            try:
+                # Query github for the latest version of Flock Agent
+                r = requests.get(
+                    "https://api.github.com/repos/firstlookmedia/flock-agent/releases/latest"
+                )
+                release = r.json()
+                latest_version = release["tag_name"].lstrip("v")
+                self.c.log(
+                    "Daemon",
+                    "autoupdate_loop",
+                    f"installed version: {self.c.version}, latest version: {latest_version}",
+                )
+                if latest_version <= self.c.version:
+                    continue
+
+                # Find the pkg asset
+                url = None
+                filename = None
+                for asset in release["assets"]:
+                    if asset["name"].endswith(".pkg"):
+                        url = asset["browser_download_url"]
+                        filename = asset["name"]
+                        break
+
+                if not url:
+                    self.c.log("Daemon", "autoupdate_loop", "could not find .pkg file")
+                    continue
+
+                # Download the update
+                self.c.log("Daemon", "autoupdate_loop", f"downloading {url}")
+                r = requests.get(url)
```

Comment:
How big do we expect this package to get? Are we loading it all into memory here?

Reply:
The package is about 34mb, and I don't expect releases to get very much larger. So I figure loading it all in memory should be fine.